### DISCUSSION TITLE
Is it possible to impose only a left boundary condition?

### DISCUSSION THREAD

**Discussion by HannoSpreeuw** [zwicker-group/py-pde#351] [Q&A] [answered]
I am trying to mimic work from a paper where only a left boundary condition is specified for a 1D depth grid, i.e. only a condition at the surface is specified for two particular PDEs and not at the bottom. 

Btw, the other three PDEs - this is about solving a set of five coupled PDEs - do have boundary conditions specified at the bottom.

Is it possible to do this in py-pde?
It seems you always have to specify both a left and right boundary condition.

**Comment by david-zwicker** [comment, marked as answer]
You are correct that `py-pde` always requires boundary conditions on both sides since this was most natural for the use cases for which we developed the package. However, if you use one-sided approximations in the finite-difference scheme (e.g., `forward` or `backward` derivatives), one of the boundary condition will not be used, so you can set it to an arbitrary value. It might be a good idea to use absurdly large values or even `np.nan` to ensure that the boundary condition does not spill into the simulation.

**Reply by HannoSpreeuw** [reply]
But, to be clear, is there no way to circumvent this, to impose only a left or a right boundary condition?

**Reply by david-zwicker** [reply]
No currently, there is no way to do this. I guess we could implement a no-op boundary condition, but I still don't really see the use case. If you only use one-sided derivatives, you know that one of the boundary conditions will not be used and so the chosen condition will not affect the simulation.

**Reply by HannoSpreeuw** [reply]
Yes, I understand, but a one-sided derivative is only first order accurate.

**Reply by NiklasHohmann** [reply]
What about the case of a linear transport equation, whose solution is a wave propagating into one direction? Analytically there should only be boundary condition specified on the upwind side. I know that numerically this should be approached with an upwind(-biased) scheme, but for any scheme that uses information from the downwind side, the bc will spill into the integration.

**Reply by david-zwicker** [reply]
I agree with your statements. What is your concrete question?

**Reply by NiklasHohmann** [reply]
I just wanted to make sure I understood correctly, thanks for your answer!

**Comment by aminih01** [comment]
It might not be relevant, but how can we separately define left/right boundaries for different fields in a coupled problem? I am trying to set different boundary types by saying bc_x_left = {"value" :0} for one field, but I don't know how to prevent that from being applied to both fields as I need bc_x_left to be set to derivative for my second field.

**Reply by david-zwicker** [reply]
This is certainly possible. If you use the `PDE` class, you need to specify boundary conditions using the `bc_ops` argument; see [the documentation](https://py-pde.readthedocs.io/en/latest/packages/pde.pdes.pde.html).
